### PR TITLE
Rate limit logins by username and IP address

### PR DESCRIPTION
**Pull request by Dantemss**
Currently set to 100 username login failures and 10,000 IP login failures

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -129,19 +129,21 @@ def help
 
   # Omniauth failure endpoint
   def failure
-    security_log :sign_in_failed, reason: params[:message]
```
Comment (jpslav) [inline app/controllers/sessions_controller.rb]: Why did this line go away?

File: spec/features/rate_limit_logins_spec.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      expect(page).to have_content('You have made too many login attempts recently.')
+
+      visit '/'
+      expect_sign_in_page
+
+      fill_in 'Username', with: 'user'
+      fill_in 'Password', with: 'password'
+      click_button 'Sign in'
+      expect(page).to have_content('You have made too many login attempts recently.')
+    end
```
Comment (jpslav) [inline spec/features/rate_limit_logins_spec.rb]: Can you timecop and show that the lockout is lifted after an hour?

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (Dantemss) [inline app/controllers/sessions_controller.rb]: Moved to the CustomIdentity strategy. Basically CustomIdentity redirects you here, but a bot could just not follow the redirect and prevent the log from being created. So I moved it to when the password is actually checked.

File: spec/features/rate_limit_logins_spec.rb (hunk 2 above)
Comment (Dantemss) [inline spec/features/rate_limit_logins_spec.rb]: Oh sure forgot about that test :)

**Pull request merged**